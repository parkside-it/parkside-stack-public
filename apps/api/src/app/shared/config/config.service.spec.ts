import { Test } from '@nestjs/testing';
import { ConfigService } from '@psb-shared';
import { parse as envParse } from 'dotenv';
import { readFileSync } from 'fs';

describe('ConfigService', () => {
  let configService: ConfigService;

  beforeEach(async () => {
    const env = envParse(readFileSync(`.env.defaults`));
    const module = await Test.createTestingModule({
      providers: [
        {
          provide: ConfigService,
          useValue: new ConfigService(env),
        },
      ],
    }).compile();

    configService = module.get<ConfigService>(ConfigService);
  });

  it('should match snapshot', () => {
    expect(configService).toMatchSnapshot();
  });
});
