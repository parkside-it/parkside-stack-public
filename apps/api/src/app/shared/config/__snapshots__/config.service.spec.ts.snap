// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigService should match snapshot 1`] = `
ConfigService {
  "envConfig": Object {
    "ACCESS_TOKEN_EXPIRATION_SECONDS": "3600",
    "API_PATH": "api",
    "API_VERSION": "0.1",
    "BASE_URL": "http://localhost:4200",
    "DB_DATABASE": "test",
    "DB_HOST": "localhost",
    "DB_PASSWORD": "password",
    "DB_PORT": "3306",
    "DB_USER": "root",
    "EMAIL_DEFAULT_SENDER": "Firstname Lastname <user@example.com>",
    "EMAIL_SMTP_SECRET": "some_secret",
    "EMAIL_TEMPLATE_DIRECTORY": "apps/api/src/templates/email",
    "ENV": "development",
    "JWT_SECRET": "jwtsecret",
    "PASSWORD_RESET_TOKEN_EXPIRATION_SECONDS": "3600",
  },
}
`;
